src/client/oicp/CpoOICPClient.ts

```diff
@@ -0,0 +1,1272 @@
+        detailedMessages: { response: response }
+      });
+    }
+  }
+
+  async stopSession(transaction: Transaction): Promise<void> {
+    if (!transaction.oicpData) {
+      throw new BackendError({
+        source: transaction.chargeBoxID,
+        action: ServerAction.OICP_PUSH_SESSIONS,
+        message: `OICP data does not exists on Session ID '${transaction.id}'`,
+        module: MODULE_NAME, method: 'stopSession',
+      });
+    }
+    if (!transaction.oicpData.session) {
+      throw new BackendError({
+        source: transaction.chargeBoxID,
+        action: ServerAction.OICP_PUSH_SESSIONS,
+        message: `OICP Session data does not exists on Session ID '${transaction.id}'`,
+        module: MODULE_NAME, method: 'stopSession',
+      });
+    }
+    if (!transaction.stop) {
+      throw new BackendError({
+        source: transaction.chargeBoxID,
+        action: ServerAction.OICP_PUSH_SESSIONS,
+        message: `OICP Session ID '${transaction.oicpData.session.id}' (ID '${transaction.id}') not yet stopped`,
+        module: MODULE_NAME, method: 'stopSession',
+      });
+    }
+
+    transaction.oicpData.session.kwh = transaction.stop.totalConsumptionWh / 1000;
+    transaction.oicpData.session.total_cost = transaction.stop.roundedPrice > 0 ? transaction.stop.roundedPrice : 0;
+    transaction.oicpData.session.end_datetime = transaction.stop.timestamp;
+    transaction.oicpData.session.last_updated = transaction.stop.timestamp;
+    transaction.oicpData.session.status = OICPSessionStatus.COMPLETED;
+    // If (transaction.lastConsumption) {
+    //   transaction.oicpData.session.meterValueInBetween.push(transaction.lastConsumption.value);
+    // }
+
+    // Log
+    Logging.logDebug({
+      tenantID: this.tenant.id,
+      action: ServerAction.OICP_PUSH_SESSIONS,
+      message: `Stop OICP Transaction ID '${transaction.oicpData.session.id}' (ID '${transaction.id}') to Hubject`,
+      module: MODULE_NAME, method: 'stopSession',
+      detailedMessages: { payload: transaction.oicpData.session }
+    });
+    // Call Hubject
+    await this.sendChargingNotificationEnd(transaction);
+    const response = await this.authorizeStop(transaction);
+
+    Logging.logDebug({
+      tenantID: this.tenant.id,
+      action: ServerAction.OICP_PUSH_SESSIONS,
+      message: `Push OICP Transaction ID '${transaction.oicpData.session.id}' (ID '${transaction.id}') response retrieved from Hubject`,
+      module: MODULE_NAME, method: 'stopSession',
+      detailedMessages: { response: response }
+    });
+  }
+
+  /**
+   * Send all EVSEs
+   */
+  async sendEVSEs(processAllEVSEs = true, actionType?: OICPActionType): Promise<OICPJobResult> {
+    if (!actionType) {
+      actionType = OICPActionType.fullLoad;
+    }
+    // Result
+    const sendResult = {
+      success: 0,
+      failure: 0,
+      total: 0,
+      logs: [],
+      chargeBoxIDsInFailure: [],
+      chargeBoxIDsInSuccess: []
```

Comment:
Check the latest changes made on OCPI code, that fields are not part of OICPJobResult. Just add the right type to sendResult to fix the typing issue.

Reply:
done